twill-yarn/src/main/java/org/apache/twill/internal/appmaster/RunningContainers.java

```diff
@@ -113,9 +117,11 @@ public Integer apply(BitSet input) {
   private final Location applicationLocation;
   private final Set<String> runnableNames;
   private final Map<String, Map<String, String>> logLevels;
+  private final Map<String, Integer> maxRetries;
```

Comment:
On further thought, I think if we track the restarts per instance id then it would simplify the contract for max retries. This will be consistent with what the javadoc says too -
```
Sets the maximum number of times (per instance) a runnable will be retried if it exits without success. The default behavior is to retry indefinitely.
```